### ISSUE TITLE
Templated @record types fail inference with other templates

### ISSUE THREAD

**Issue by SLaks** [google/closure-compiler#1836]
I have an `@record` type with a template parameter which is not yet used.

I perform a complicated dance with two other template types (in my actual code, this is from a base Model class), and I get an incorrect compiler error:

```js
/** @constructor @template T */
function EventType() {}


/** @struct @record @template MODEL */
function ChangeEventArgs() {};
/** @const {number} */
ChangeEventArgs.prototype.reason;


/** @constructor @template U */
function MyClass(){}

/**
 * @param {!EventType<T>} type
 * @param {T} args
 * @template T
 */
MyClass.prototype.notify = function(type, args){}

/** @const {!EventType<!ChangeEventArgs<U>>} */
MyClass.prototype.eventType = new EventType();

MyClass.prototype.test = function() {
  this.notify(this.eventType, {reason: 0});
};
```

```
input0:25: WARNING - actual parameter 1 of MyClass.prototype.notify does not match formal parameter
found   : EventType<ChangeEventArgs<U>>
required: EventType<(ChangeEventArgs<U>|{reason: number})>
  this.notify(this.eventType, {reason: 0});
              ^
```

https://closure-compiler-debugger.appspot.com/#input0%3D%252F**%2520%2540constructor%2520%2540template%2520T%2520*%252F%250Afunction%2520EventType()%2520%257B%257D%250A%250A%250A%252F**%2520%2540struct%2520%2540record%2520%2540template%2520MODEL%2520*%252F%250Afunction%2520ChangeEventArgs()%2520%257B%257D%253B%250A%252F**%2520%2540const%2520%257Bnumber%257D%2520*%252F%250AChangeEventArgs.prototype.reason%253B%250A%250A%250A%252F**%2520%2540constructor%2520%2540template%2520U%2520*%252F%250Afunction%2520MyClass()%257B%257D%250A%250A%252F**%250A%2520*%2520%2540param%2520%257B!EventType%253CT%253E%257D%2520type%250A%2520*%2520%2540param%2520%257BT%257D%2520args%250A%2520*%2520%2540template%2520T%250A%2520*%252F%250AMyClass.prototype.notify%2520%253D%2520function(type%252C%2520args)%257B%257D%250A%250A%252F**%2520%2540const%2520%257B!EventType%253C!ChangeEventArgs%253CU%253E%253E%257D%2520*%252F%250AMyClass.prototype.eventType%2520%253D%2520new%2520EventType()%253B%250A%250AMyClass.prototype.test%2520%253D%2520function()%2520%257B%250A%2520%2520this.notify(this.eventType%252C%2520%257Breason%253A%25200%257D)%253B%250A%257D%253B%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26CHECK_SYMBOLS%3D1%26CHECK_TYPES%3D1%26CLOSURE_PASS%3D1%26LANG_IN_IS_ES6%3D1%26MISSING_PROPERTIES%3D1%26PRESERVE_TYPE_ANNOTATIONS%3D1%26PRETTY_PRINT%3D1%26TRANSPILE%3D1

**Comment by dimvar** [comment]
FYI, this works correctly in the new type inference.
